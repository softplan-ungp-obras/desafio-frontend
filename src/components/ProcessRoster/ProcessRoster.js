import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ProcessSummary from '../ProcessSummary/ProcessSummary';

import { List, ListItem, NoResults } from './ProcessRoster.styles';

const ProcessRoster = ({ processList, reduced }) => {
  return (
    <Fragment>
      {processList.length > 0 ? (
        <List isListReduced={reduced}>
          {processList.map(process => {
            return (
              <ListItem key={process.id}>
                <ProcessSummary isListReduced={reduced} process={process} />
              </ListItem>
            );
          })}
        </List>
      ) : (
        <NoResults>Nenhum resultado para sua busca</NoResults>
      )}
    </Fragment>
  );
};

const mapStateToProps = ({ processList }) => {
  return {
    processList: processList.data
  };
};

ProcessRoster.propTypes = {
  processList: PropTypes.array,
  reduced: PropTypes.bool
};

export default connect(mapStateToProps)(ProcessRoster);
