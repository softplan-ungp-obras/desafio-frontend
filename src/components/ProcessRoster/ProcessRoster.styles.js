import styled from 'styled-components';
import device from '../../helpers/breakpoints';

export const List = styled.ul`
  list-style: none;
  margin: ${props => (props.isListReduced ? '0 20px 0 130px' : '0 0 0 130px')};
  padding: 0;
  width: ${props => (props.isListReduced ? '30%' : '75%')};

  @media ${device.mobile} {
    margin: 0 20px;
    width: auto;
  }
`;

export const ListItem = styled.li`
  margin: 0 0 20px 0;

  @media ${device.mobile} {
    margin-right: 15px;
    max-width: 100%;
  }
`;

export const NoResults = styled.div`
  text-align: center;
  width: 50%;
`;
