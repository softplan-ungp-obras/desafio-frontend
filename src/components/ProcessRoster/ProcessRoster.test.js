import React from 'react';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';

import ProcessRoster from './ProcessRoster';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const mockLocation = {
  state: {
    processId: 'c05a14b8-9f98-4e4d-908c-050e0c46a7f2'
  }
};

describe('ProcessRoster', () => {
  it('should render the component correctly', () => {
    // given
    const mockProcessList = {
      data: [
        {
          id: 'c05a14b8-9f98-4e4d-908c-050e0c46a7f2',
          numero: 'SOFT 2020/00001',
          entrada: '10/02/2019',
          interessados: ['João da Silva'],
          descricao:
            'Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2018',
          assunto: 'Licença'
        },
        {
          id: 'c05a14b8-9f98-4e4d-908c-050e0c46a7f3',
          numero: 'SOFT 2020/00001',
          entrada: '10/02/2019',
          interessados: ['João da Silva'],
          descricao:
            'Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2020',
          assunto: 'Licença'
        }
      ]
    };
    const store = mockStore({ processList: { ...mockProcessList } });

    // when
    const component = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[mockLocation]}>
          <ProcessRoster />
        </MemoryRouter>
      </Provider>
    );

    const { getAllByText } = component;

    // then
    expect(component).toMatchSnapshot('regular_list');
    expect(getAllByText('SOFT 2020/00001')).toHaveLength(2);
  });

  it('should render no results component', () => {
    // given
    const mockProcessList = {
      data: []
    };
    const store = mockStore({
      processList: {
        ...mockProcessList
      }
    });

    // when
    const component = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[mockLocation]}>
          <ProcessRoster />
        </MemoryRouter>
      </Provider>
    );

    const { getAllByText } = component;

    // then
    expect(component).toMatchSnapshot('no_results');
    expect(getAllByText('Nenhum resultado para sua busca')).toHaveLength(1);
  });
});
