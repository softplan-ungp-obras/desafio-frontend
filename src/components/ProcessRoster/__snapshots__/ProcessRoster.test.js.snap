// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessRoster should render no results component: no_results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bZQynM jtjfSu"
      >
        Nenhum resultado para sua busca
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bZQynM jtjfSu"
    >
      Nenhum resultado para sua busca
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProcessRoster should render the component correctly: regular_list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-ifAKCX eXALfk"
      >
        <li
          class="sc-EHOje dSogAF"
        >
          <div
            class="sc-bdVaJa cWtrlz"
          >
            <a
              href="/process-detail/process=c05a14b8-9f98-4e4d-908c-050e0c46a7f2"
              style="align-items: center; display: flex; flex-wrap: wrap; text-decoration: none; width: 100%;"
            >
              <div
                class="sc-bwzfXH EtGxv"
              >
                <img
                  alt=""
                  src="https://via.placeholder.com/80"
                />
              </div>
              <div
                class="sc-bwzfXH dGJtax"
              >
                <p
                  class="sc-htpNat cWKNlF"
                >
                  Número
                </p>
                <p
                  class="sc-bxivhb Cdfip"
                >
                  SOFT 2020/00001
                </p>
              </div>
              <div
                class="sc-bwzfXH EtGxv"
              >
                <p
                  class="sc-htpNat cWKNlF"
                >
                  Assunto
                </p>
                <p
                  class="sc-bxivhb Cdfip"
                >
                  Licença
                </p>
              </div>
              <div
                class="sc-bwzfXH dGJtax"
              >
                <p
                  class="sc-htpNat cWKNlF"
                >
                  Interessado
                </p>
                <p
                  class="sc-bxivhb Cdfip"
                >
                  João da Silva
                </p>
              </div>
              <div
                class="sc-bwzfXH EtGxv"
              >
                <p
                  class="sc-htpNat cWKNlF"
                >
                  Descrição
                </p>
                <p
                  class="sc-bxivhb Cdfip"
                >
                  Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2018
                </p>
              </div>
            </a>
          </div>
        </li>
        <li
          class="sc-EHOje dSogAF"
        >
          <div
            class="sc-bdVaJa ReZNG"
          >
            <a
              href="/process-detail/process=c05a14b8-9f98-4e4d-908c-050e0c46a7f3"
              style="align-items: center; display: flex; flex-wrap: wrap; text-decoration: none; width: 100%;"
            >
              <div
                class="sc-bwzfXH EtGxv"
              >
                <img
                  alt=""
                  src="https://via.placeholder.com/80"
                />
              </div>
              <div
                class="sc-bwzfXH dGJtax"
              >
                <p
                  class="sc-htpNat diWegQ"
                >
                  Número
                </p>
                <p
                  class="sc-bxivhb kKiemp"
                >
                  SOFT 2020/00001
                </p>
              </div>
              <div
                class="sc-bwzfXH EtGxv"
              >
                <p
                  class="sc-htpNat diWegQ"
                >
                  Assunto
                </p>
                <p
                  class="sc-bxivhb kKiemp"
                >
                  Licença
                </p>
              </div>
              <div
                class="sc-bwzfXH dGJtax"
              >
                <p
                  class="sc-htpNat diWegQ"
                >
                  Interessado
                </p>
                <p
                  class="sc-bxivhb kKiemp"
                >
                  João da Silva
                </p>
              </div>
              <div
                class="sc-bwzfXH EtGxv"
              >
                <p
                  class="sc-htpNat diWegQ"
                >
                  Descrição
                </p>
                <p
                  class="sc-bxivhb kKiemp"
                >
                  Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2020
                </p>
              </div>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-ifAKCX eXALfk"
    >
      <li
        class="sc-EHOje dSogAF"
      >
        <div
          class="sc-bdVaJa cWtrlz"
        >
          <a
            href="/process-detail/process=c05a14b8-9f98-4e4d-908c-050e0c46a7f2"
            style="align-items: center; display: flex; flex-wrap: wrap; text-decoration: none; width: 100%;"
          >
            <div
              class="sc-bwzfXH EtGxv"
            >
              <img
                alt=""
                src="https://via.placeholder.com/80"
              />
            </div>
            <div
              class="sc-bwzfXH dGJtax"
            >
              <p
                class="sc-htpNat cWKNlF"
              >
                Número
              </p>
              <p
                class="sc-bxivhb Cdfip"
              >
                SOFT 2020/00001
              </p>
            </div>
            <div
              class="sc-bwzfXH EtGxv"
            >
              <p
                class="sc-htpNat cWKNlF"
              >
                Assunto
              </p>
              <p
                class="sc-bxivhb Cdfip"
              >
                Licença
              </p>
            </div>
            <div
              class="sc-bwzfXH dGJtax"
            >
              <p
                class="sc-htpNat cWKNlF"
              >
                Interessado
              </p>
              <p
                class="sc-bxivhb Cdfip"
              >
                João da Silva
              </p>
            </div>
            <div
              class="sc-bwzfXH EtGxv"
            >
              <p
                class="sc-htpNat cWKNlF"
              >
                Descrição
              </p>
              <p
                class="sc-bxivhb Cdfip"
              >
                Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2018
              </p>
            </div>
          </a>
        </div>
      </li>
      <li
        class="sc-EHOje dSogAF"
      >
        <div
          class="sc-bdVaJa ReZNG"
        >
          <a
            href="/process-detail/process=c05a14b8-9f98-4e4d-908c-050e0c46a7f3"
            style="align-items: center; display: flex; flex-wrap: wrap; text-decoration: none; width: 100%;"
          >
            <div
              class="sc-bwzfXH EtGxv"
            >
              <img
                alt=""
                src="https://via.placeholder.com/80"
              />
            </div>
            <div
              class="sc-bwzfXH dGJtax"
            >
              <p
                class="sc-htpNat diWegQ"
              >
                Número
              </p>
              <p
                class="sc-bxivhb kKiemp"
              >
                SOFT 2020/00001
              </p>
            </div>
            <div
              class="sc-bwzfXH EtGxv"
            >
              <p
                class="sc-htpNat diWegQ"
              >
                Assunto
              </p>
              <p
                class="sc-bxivhb kKiemp"
              >
                Licença
              </p>
            </div>
            <div
              class="sc-bwzfXH dGJtax"
            >
              <p
                class="sc-htpNat diWegQ"
              >
                Interessado
              </p>
              <p
                class="sc-bxivhb kKiemp"
              >
                João da Silva
              </p>
            </div>
            <div
              class="sc-bwzfXH EtGxv"
            >
              <p
                class="sc-htpNat diWegQ"
              >
                Descrição
              </p>
              <p
                class="sc-bxivhb kKiemp"
              >
                Solicitação de licença-prêmio referente ao período 02/06/2015 - 01/06/2020
              </p>
            </div>
          </a>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
